components/elm/src/data_types/VegetationPropertiesType.F90

```diff
@@ -305,10 +308,12 @@ subroutine veg_vp_init(this)
     allocate(this%tc_stress    )
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
+    allocate(this%nonvascular(0:numpft))                         ; this%nonvascular(:)           =huge(1)
+    allocate(this%nfixer(0:numpft))                              ; this%nfixer(:)                =huge(1)
 
     do m = 0,numpft
 
-       if (m <= ntree) then
+       if (woody(m) == 1) then
           this%tree(m) = 1
```

Comment:
If tree is not being used anywhere, perhaps it can be removed to avoid confusion